src/api/unified/symbol_manager.hpp

```diff
@@ -23,6 +23,7 @@
 namespace unified {
 
 const int NUM_BACKENDS = 3;
+const int MAX_BKND_HANDLES = 10;
```

Comment:
The docs say max 7 binaries can be loaded. How this calculation work ?

Reply:
3 are reserved for the default backend libraries that the `AFSymbolManager` loads (regardless of how many backends does the machine or the build actually supports).

Reply:
what happens when loading of default backends fails ?

Reply:
If the program doesn't add and set custom libraries, the arrayfire functions would fail ("Failed to load dynamic library") because the handles are null . This is the behavior even before this change. However, if the user does add and set custom libraries, then everything still works. I simulated this by making a small change in the `AFSymbolManager` constructor:

``` cpp
    // Decremeting loop. The last successful backend loaded will be the most
    // prefered one.
    for (int i = NUM_BACKENDS - 1; i >= 0; i--) {
        int backend          = order[i] >> 1;  // 2 4 1 -> 1 2 0
        // bkndHandles[backend] = openDynLibrary(order[i]); // <--
        if (bkndHandles[backend]) {
            activeHandle  = bkndHandles[backend];
            activeBackend = (af_backend)order[i];
            numBackends++;
            backendsAvailable += order[i];
        }
    }
```

This application code still works:
```cpp
    string lib_path = "/home/mark/Documents/arrayfire-3.6.3/build/src/backend/cuda/libafcuda.so";
    af_add_backend_library(lib_path.c_str());
    af_set_backend_library(0);
    array a = randu(3, 3);
    array b = constant(5, 3, 3);
    array c = a + b;
    info();
    af_print(a);
    af_print(b);
    af_print(c);
```

Reply:
Nah, you got me wrong there. I asked what happens if the defaults loading fails - the first three you are reserving for the current (lets refer to the ones that exist next to libaf as current)  binaries. The custom handles offset (`numBackendHandles`) won't change I guess.

I assume, unless a set is called using custom handle index, our default backend-order-preference prevails, is that correct ?

Reply:
Oh whoops, thanks for the clarification. Yes, the default backend order prevails. The code for setting that doesn't change